src/internal.rs

```diff
@@ -37,6 +41,58 @@ impl Default for FileTypes {
     }
 }
 
+enum SizeLimitType {
+    Max,
+    Min,
+}
+
+pub struct SizeFilter {
+    size: u64,
+    limit_type: SizeLimitType,
+}
+
+impl SizeFilter {
+    pub fn is_within(&self, size: u64) -> bool {
+        match self.limit_type {
+            SizeLimitType::Max => size <= self.size,
+            SizeLimitType::Min => size >= self.size,
+        }
+    }
+}
+
+const KILO: u64 = 1024;
+const MEGA: u64 = KILO * 1024;
+const GIGA: u64 = MEGA * 1024;
+const TERA: u64 = GIGA * 1024;
+
+impl<'a> From<&'a str> for SizeFilter {
+    /// Create the `SizeFilter` from the given `&str`.
+    /// It is imperative that the incoming value has been validated for
+    /// proper format.
+    fn from(s: &str) -> Self {
+        let captures = SIZE_CAPTURES.captures(s).unwrap();
+        let limit = match captures.get(1).map_or("+", |m| m.as_str()) {
+            "+" => SizeLimitType::Min,
+            _ => SizeLimitType::Max,
+        };
+
+        let quantity = captures.get(2).unwrap().as_str().parse::<u64>().unwrap();
```

Comment:
If we move the verification to this function, we could replace all the `.unwrap()` calls by `?`.